File: 3617-find-the-original-typed-string-i/3617-find-the-original-typed-string-i.go
func possibleStringCount(word string) int {
    n := len(word)
    ans := 0

    for i := 0; i < n; i++ {
        cnt := 0
        for i < n-1 && word[i] == word[i+1] {
            i++
            cnt++
        }
        ans += cnt
    }

    return ans + 1
}

File: 3617-find-the-original-typed-string-i/3617-find-the-original-typed-string-i.java
class Solution {
    public int possibleStringCount(String word) {
        int n = word.length();
        int ans = 0;
        for(int i = 0; i < n; i++) {
            int cnt = 0;
            while(i<n-1 && word.charAt(i)==word.charAt(i+1)){
                i++;
                cnt++;
            }
            ans += cnt;
        }
        return ans + 1;
    }
}